Complete RFP (#9)
stephenneale/seng371-group16: as-clients/rfp.md

| v1.0    | 2023-01-24      | Client team            | Initial drafting                          |

## Table of Contents

(1.0) Product Overview  
(2.0) Project Objectives  
(3.0) Current System(s)  
(4.0) Intended Users of the System  
(5.0) Known interactions with systems within or outside the client organization  
(6.0) Known constraints to development  
(7.0) Project Schedule  
(8.0) Project team  
(9.0) Glossary of terms  

## (1.0) Product Overview

We the people of “Atlantis” have resurfaced from the depths of the ocean. We have our own currency and bank but need an online banking system to reconnect with the world. The issues we have are the following:

- Unable to e-transfer other banking accounts
- Unable to view avaible funds
- No online payment methods
- The city of Atlantis is growing and visiting the bank is not scalable. Customers having to go to the bank just to view the transaction history is causing problems. 

We predict that making an online banking system will solve these issues and increase customer satisfaction for all their banking needs. 

## (2.0) Project Objectives

> The (name of bank) online banking system needs to fulfill the following needs

Author: We need to replace this. We can even avoid mentioning a bank name at all, see suggestion
```suggestion
The online banking system needs to fulfil the following needs
```

- All users can view funds and transaction history, including but not limited too: withdrawls, purchases, and deposits. 
- Users can view their credit card information, including but not limited too, credit card number, security code and credit limits.
- Users can create investment and savings accounts

> - The sytem should be available 24/7

Author: ```suggestion
- The system should be available 24/7
```

- The system must be compatible with the Chrome browser as it is the only browser avaible to Atlanteans.

> - The system must supports up-to-date security measures

Author: ```suggestion
- The system must support up-to-date security measures
```

## (3.0) Current System(s)

> The (name of bank) has a physcial bank and no online presence.

Author: See above
```suggestion
We have a physical bank and no online presence.
```

## (4.0) Intended users of the system

There are two intended users of the sytem
- Bank Managers

> - Users of the bank
> 
>  Users of the bank are the primary users and will be able to manage all of their banking needs through the chrome web browser. Users should be able to interact with the software through their personal computers or on mobile phones.

Author: I think we could replace "users" with "clients" here, since "users" of the system are both clients and managers, while "clients" is more specific like the people who open accounts and use the bank's services as customers. What do you guys think?
```suggestion
- Bank clients

 Clients of the bank are the primary users and will be able to manage all of their banking needs through the chrome web browser. Clients should be able to interact with the software through their personal computers or on mobile phones.
```

>  Bank managers are the secondary users of the bank and should be able to their bank managing needs (idk what bank managers need to do)

Author: idk either lol. let me freestyle this…
```suggestion
 Bank managers are the secondary users of the system and should be able to perform their functions effectively, including viewing client accounts and making changes, subject to access restrictions appropriate for their stations.
```

systems of Planet Earth. This is so that our clients can fully participate in the global economy.
Concretely, the new banking system should integrate effectively with the SWIFT network.

Additionally, the system should effectively integrate with the Internet so that customers travelling
at home or around the world can access the system wherever they are.

## (6.0) Known constraints to development

Given that banking is a tightly regulated field that deals with high-value sensitive information,
every effort must be taken to ensure that the system is not vulnerable to security attacks and
that system integrity is prioritized.

Additionally, in the interest of cost saving, we are focussing on the development of a system that
is compatible with the Chrome web browser, compatibility with other browsers given little priority.
This is because Chrome is used by the vast majority of internet users in Atlantis. Additionally,
a mobile app is not prioritized at this time. The web version suffices.

## (7.0) Project Schedule

TBD

## (8.0) Project team

| Name          | Role                        |
| ---           | ---                         |

> | Stephen Neale |                             |

Author: @stephenneale be sure to fill in your role 😃

| Drew Petten   | Head of project design      |
| Matt Lebl     | Co-op student               |

> | Samuel Gao    |                             |

Author: @SamurayG be sure to fill in your role as well 😉
